stix/idl/processing/coordinates/stx_location4spectroscopy.pro

```diff
@@ -0,0 +1,60 @@
+;---------------------------------------------------------------------------
+;+
+; :project:
+;       STIX
+;
+; :name:
+;       stx_location4spectroscopy
+;
+; :description:
+;    This converts a flare location from Helioprojective Cartesian coordinate frame at Solar Orbiter (HPC) to STIX coordinate frame (STX)
+;    for calculation of grid transmission. 
+;    
+; :categories:
+;    spectroscopy, transmission 
+;
+; :keywords:
+;    flare_location_hpc : in, type="2 element float array"
+;               the location of the flare (X,Y) in Helioprojective Cartesian coordinates as seen from Solar Orbiter [arcsec]
+;               If input the HPC flare location will be transformed to the STIX coordinate frame
+;               Otherwise a dummy array of [Nan, Nan] will be passed out to the given named variable
+;               
+;    aux_fits_file :in, required if flare_location_hpc is passed in, type="string"
+;                the path of the auxiliary ephemeris FITS file to be read."
+;    
+;    time_range: in, type="string array"
+;                array containing the UTC start time and end time of the spectrogram
```

Comment:
Is this Earth UT, Solar Orbiter UT or can it be any time reference, as long as the user sets time_shift as needed?

Reply:
This is directly passed to stx_create_auxiliary_data.pro so it should be the same reference. I would think that it ought to be Earth UT @paolomassa can you confirm?

Reply:
The time considered within `stx_create_auxiliary_data.pro` is read from an L2 ephemeris fits file. Therefore, it should be in Solar Orbiter UT. I would ask @FredSchuller to confirm this point in order to be 100% sure

Reply:
Yes all times in L1 and above are Solar Orbiter UT